File: flutter_app/lib/models/project_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'customer_model.dart';

part 'project_model.freezed.dart';
part 'project_model.g.dart';

enum EProjectState {
  planned,
  inProgress,
  finished;
}

@freezed
abstract class ProjectModel with _$ProjectModel {
  const factory ProjectModel({
    int? id,
    required String name,
    String? description,
    required EProjectState state,
    required bool isPublic,
    required String startDate,
    required String endDate,
    double? budget,
    int? estimatedHours,
    required CustomerModel customer,
    int? customerId
  }) = _ProjectModel;

  factory ProjectModel.fromJson(Map<String, dynamic> json) => _$ProjectModelFromJson(json);
}

File: angular_app/src/app/models/project.model.ts
import {EProjectState} from '../enums/e-project-state';
import {CustomerModel} from './customer.model';

export interface ProjectModel {
  id?: number
  name: string
  description?: string
  state: EProjectState
  isPublic: boolean
  startDate: string
  endDate: string
  budget?: number
  estimatedHours?: number
  customer: CustomerModel
  customerId?: number
  index: number
}

export interface ProjectStateList {
  state: EProjectState
  projects: ProjectModel[]
}
